File: algorithms/js/removeDuplicatesFromSortedArray/removeDuplicatesFromSortedArray.js
// Source : https://leetcode-cn.com/problems/remove-duplicates-from-sorted-array/
// Author : Tian Zhuo
// Date   : 2018-11-23

/**
 * @param {number[]} nums
 * @return {number}
 */
var removeDuplicates = function(nums) {
    if(nums.length == 0) return 0;
    let i = 1;
    while(i < nums.length){
        if(nums[i - 1] == nums[i]){
            nums.splice(i, 1);
        }else{
            i++;
        }
    }
    return nums.length;
};

/**
 * @param {number[]} nums
 * @return {number}
 */
var removeDuplicates1 = function(nums) {
    let i = 0;
    
    for(let j = 0; j < nums.length; j++) {
        if(nums[j] != nums[i]){
            i++;
            nums[i] = nums[j];
        }
    }
    
    return i+1;
};

File: src/main/java/removeDuplicatesFromSortedArray/RemoveDuplicatesFromSortedArray.java
/*
 * @Source: https://leetcode-cn.com/problems/remove-duplicates-from-sorted-array/
 * @Author: Tian Zhuo
 * @Date: 2018-12-22 14:59:54
 */
package removeDuplicatesFromSortedArray;

public class RemoveDuplicatesFromSortedArray {
    public int removeDuplicates(int[] nums) {
        if(nums.length == 0) return 0;
        int i = 0;
        for (int j = 1; j < nums.length; j++) {
            if(nums[j] != nums[i]){
                i++;
                nums[i] = nums[j];
            }
        }
        
        return i + 1;
    }
}